### PR TITLE
Fix reporting in e2e tests

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

#15983 updated the e2e tests to use FLUID_TEST_MULTIREPORT, but that option previously didn't respect the `testReportPrefix` setting we use to make sure different e2e test configs produces different test result files.
As a result, the tinylicious test results were clobbering those from local server, which makes test failures in local server difficult to diagnose.

This fixes our reporting output files to once again be distinct, which will hopefully make the intermittent failures we're seeing easy to diagnose (it's likely just a flaky test)

### CODE REVIEW

**Comment by Abe27342** [comment]
There's probably some cleanup that can be done of FLUID_TEST_REPORT. didn't want to attempt it here though.

File: packages/test/mocha-test-setup/mocharc-common.js [hunk 1]
```diff
@@ -78,7 +78,14 @@ function getFluidTestMochaConfig(packageDir, additionalRequiredModules, testRepo
...
 		config["reporter"] = `mocha-multi-reporters`;
-		config["reporter-options"] = [`configFile=${path.join(__dirname, "test-config.json")}`];
+		// See https://www.npmjs.com/package/mocha-multi-reporters#cmroutput-option
+		const outputFilePrefix = testReportPrefix !== undefined ? `${testReportPrefix}-` : "";
+		config["reporter-options"] = [
+			`configFile=${path.join(
+				__dirname,
+				"test-config.json",
+			)},cmrOutput=xunit+output+${outputFilePrefix}:mocha-json-output-reporter+output+${outputFilePrefix}`,
+		];
```
Comment (NicholasCouri) [inline packages/test/mocha-test-setup/mocharc-common.js:88]: Should we add a console log for the reporter-options content  ?

**Review by NicholasCouri — APPROVED** [review]

File: packages/test/mocha-test-setup/mocharc-common.js (hunk 1 above)
Comment (Abe27342) [reply packages/test/mocha-test-setup/mocharc-common.js:88]: sure. I think that's more useful than the console.log line below anyway.

**Review by andre4i — APPROVED** [review]

File: packages/test/mocha-test-setup/mocharc-common.js [hunk 2]
```diff
@@ -78,8 +78,17 @@ function getFluidTestMochaConfig(packageDir, additionalRequiredModules, testRepo
 
 	if (process.env.FLUID_TEST_MULTIREPORT === "1") {
 		config["reporter"] = `mocha-multi-reporters`;
-		config["reporter-options"] = [`configFile=${path.join(__dirname, "test-config.json")}`];
-		console.log(`configFile=${path.join(__dirname, "test-config.json")}`);
+		// See https://www.npmjs.com/package/mocha-multi-reporters#cmroutput-option
+		const outputFilePrefix = testReportPrefix !== undefined ? `${testReportPrefix}-` : "";
+		console.log(
+			`Writing test results relative to package to nyc/${outputFilePrefix}junit-report.xml and nyc/${outputFilePrefix}junit-report.json`,
+		);
```
Comment (alexvy86) [inline packages/test/mocha-test-setup/mocharc-common.js:85]: Can we create a follow-up item to make this actually tracks what `test-config.json` says (or points out that the actual filenames are what that file says they'll be)?

**Pull request merged**